.github/report-flaky-tests/index.js

```diff
@@ -0,0 +1,249 @@
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' ).promises;
+const path = require( 'path' );
+const github = require( '@actions/github' );
+const core = require( '@actions/core' );
+const unzipper = require( 'unzipper' );
+const { formatResultsErrors } = require( 'jest-message-util' );
+
+const TEST_RESULTS_LIST = {
+	open: `<!-- __TEST_RESULTS_LIST__ -->`,
+	close: `<!-- /__TEST_RESULTS_LIST__ -->`,
+};
+const TEST_RESULT = {
+	open: '<!-- __TEST_RESULT__ -->',
+	close: '<!-- /__TEST_RESULT__ -->',
+};
+
+const metaData = {
+	render: ( json ) => `<!-- __META_DATA__:${ JSON.stringify( json ) } -->`,
+	get: ( str ) => {
+		const matched = str.match( /<!-- __META_DATA__:(.*) -->/ );
+		if ( matched ) {
+			return JSON.parse( matched[ 1 ] );
+		}
+	},
+};
+
+( async function run() {
+	const token = core.getInput( 'repo-token', { required: true } );
+	const label = core.getInput( 'label', { required: true } );
+	const artifactName = core.getInput( 'artifact-name', { required: true } );
+
+	const octokit = github.getOctokit( token );
+
+	const flakyTests = await downloadReportFromArtifact(
+		octokit,
+		artifactName
+	);
+
+	if ( ! flakyTests ) {
+		return;
+	}
+
+	const { data: issues } = await octokit.rest.issues.listForRepo( {
+		...github.context.repo,
+		labels: label,
+		per_page: 100,
+	} );
+
+	for ( const flakyTestKey in flakyTests ) {
+		const [ testPath, testTitle ] = getTestPathAndTitle( flakyTestKey );
+		const issueTitle = getIssueTitle( testTitle );
+		const reportedIssue = issues.find(
+			( issue ) => issue.title === issueTitle
+		);
+		const testResults = flakyTests[ flakyTestKey ];
+		const isTrunk = getHeadBranch() === 'trunk';
+
+		if ( reportedIssue ) {
+			let body = reportedIssue.body;
+			const meta = metaData.get( body );
+
+			if ( isTrunk ) {
+				const headCommit = github.context.sha;
+				const baseCommit = meta.baseCommit || github.context.sha;
+
+				let totalCommits = 0;
+				try {
+					const { data } = await octokit.rest.repos.compareCommits( {
+						...github.context.repo,
+						base: baseCommit,
+						head: headCommit,
+						per_page: 1,
+					} );
+
+					meta.failedTimes += testResults.length;
+					meta.totalCommits = data.total_commits + 1;
```

Comment:
I'm not understanding this. Why is the number of commits since we first reported flaky tests a good approximation for total runs? Could you give me an example?

Reply:
We're only recording runs in trunk. So if we exclude manual reruns, then the total commits since the first commit that introduced the test should be the total amount of runs the test has in trunk. I can't think of a way to get the first commit that introduced the test, so the first recorded flaky commit seems like a good enough approximation for now.

Reply:
Got it that makes sense 👍